src/datasource.ts

```diff
@@ -0,0 +1,65 @@
+import { TemplateQuery } from './types';
+
+import {
+  DataSourceApi,
+  DataQueryRequest,
+  DataSourceInstanceSettings,
+} from '@grafana/ui';
+import { TimeSeries } from '@grafana/data';
+
+export class TemplateDatasource extends DataSourceApi<TemplateQuery> {
+
+  type: string;
+  url: string;
+  name: string;
+  headers: any;
+  withCredentials: boolean;
+
+  /** @ngInject */
```

Comment:
oh, it's still angular inject , right?

Reply:
Yes, React is not supported for `DatasourceApi` yet